eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala

```diff
@@ -447,12 +447,9 @@ object PaymentRequest {
     val bolt11Data = Codecs.bolt11DataCodec.decode(data).require.value
     val signature = ByteVector64(bolt11Data.signature.take(64))
     val message: ByteVector = ByteVector.view(hrp.getBytes) ++ data.dropRight(520).toByteVector // we drop the sig bytes
-    val (pub1, pub2) = Crypto.recoverPublicKey(signature, Crypto.sha256(message))
     val recid = bolt11Data.signature.last
-    val pub = if (recid % 2 != 0) pub2 else pub1
+    val pub = Crypto.recoverPublicKey(signature, Crypto.sha256(message), recid)
```

Comment:
It seems to me that we still need to validate the signature...
If we don't we can just end up recovering a public key that will simply not match the recipient, and we won't know that the invoice is completely invalid.

Reply:
The signature is always valid since we use key recovery. We could compare the recovered node id to a user-provided value but I'm not sure it's needed and it would be another PR

Reply:
That's true, in fact signature verification here doesn't make any sense.
Maybe it's worth adding a comment explaining that for future me?